prow/tide/tide.go

```diff
@@ -479,7 +479,7 @@ func accumulateBatch(presubmits map[int]sets.String, prs []PullRequest, pjs []ku
 	}
 	states := make(map[string]*accState)
 	for _, pj := range pjs {
-		if pj.Spec.Type != kube.BatchJob {
+		if pj.Spec.Type != kube.BatchJob || pj.Spec.Context == "" {
```

Comment:
Not all presubmits should have to have contexts. If a job doesn't report to GitHub it doesn't make sense to require it to supply a context.

We ignore jobs with empty contexts here because we are keying the jobs by context and we know that jobs without a context can't possibly be required so there is no point in accumulating results for them (especially since multiple jobs may not have contexts so we would conflate their results under the same key).

Reply:
> if a job doesn't report to GitHub it doesn't make sense to require it to supply a context.

If it is a presubmit there's going to be an equivalent to a github context, like a badge or bullet, etc. So this should be filled out (and ideally match the job name)

Reply:
That SGTM, but currently we do allow context to be omitted if `skip_report` is true:
https://github.com/kubernetes/test-infra/blob/7c80a99067aebd61c99e16e8ee8b88a435390fd0/prow/config/config.go#L986-L988
So if we want to start enforcing that we should update that validation function and make an announcement.
I'll remove this empty context check now though since it is just an optimization anyways.

Reply:
Maybe open a separate issue to track it.